### PR TITLE
addError() is able to translate

### PR DESCRIPTION
**Pull request by camohub**
Is there a reason why addError does not translate?

### CODE REVIEW

File: src/Forms/Form.php [hunk 1]
```diff
@@ -488,6 +488,9 @@ public function validateMaxPostSize()
 	 */
 	public function addError($message)
 	{
+		if($this->translator) {
```
Comment (dg) [inline src/Forms/Form.php]: space after `if` is missing

**Comment by dg** [comment]
There is `BaseControl::addError` too.

**Comment by camohub** [comment]
I'll make new pull request about hat BseControll::addError().

**Comment by dg** [comment]
Why? One pull request is fine

**Comment by thaeriac** [comment]
It is BC break. Now everybody must use `$form->addError($this->translator->translate('....'))`.
After that it will be translate translated string.

**Comment by dg** [comment]
In fact, it is more complicated, because addError() is called in `Nette\Forms\Rules` with already translated messages.

File: src/Forms/Form.php [hunk 2]
```diff
@@ -488,6 +488,9 @@ public function validateMaxPostSize()
 	 */
 	public function addError($message)
 	{
+		if ($this->translator) {
```
Comment (fprochazka) [inline src/Forms/Form.php]: this is wrong, you should translate on output

**Comment by fprochazka** [comment]
The current solution is perfectly fine.

If you have custom rendering of errors, use

```smarty
<div n:foreach="$form->errors as $error">
    {_$error}
</div>
```

**Comment by fprochazka** [comment]
The current solution is perfectly fine.

----

If you have custom rendering of errors, use

```smarty
<div n:foreach="$form->errors as $error">
    {_$error}
</div>
```

---

the default renderer with default rendering of errors [uses translator by default](https://github.com/nette/forms/blob/9d15bbdcf52a98fa785476da81bf0824a340e6b4/src/Forms/Validator.php#L59-L61), which means they'll be also translated automatically

```php
$form->setTranslator($translator); // profit
```

---

if you wanna add parameters and pluralization to the mix, use [Phrase](https://api.kdyby.org/class-Kdyby.Translation.Phrase.html) object.

**Comment by camohub** [comment]
Yes you are right.

**Pull request closed without merging**